### PR TITLE
Qsdk 12.0 Refresh

### PR DESCRIPTION
**Pull request by Ansuel**
Update major package to QSDK 12.0

### CODE REVIEW

**Comment by robimarko** [comment]
I am not fan of the SSDK reverts, especially since we already package socinfo cause of CPR

**Comment by Ansuel** [comment]
Should I find the commits that add the missing entry in linux-msm or we can just update the file to the latest version ?

**Comment by robimarko** [comment]
What's the actual error?
Cause, socinfo should already be the latest one from:
https://git.codelinaro.org/clo/qsdk/oss/kernel/linux-ipq-5.4/-/blob/NHSS.QSDK.12.0/include/soc/qcom/socinfo.h

**Comment by Ansuel** [comment]
Missing include not that you ask... so could be that i'm my fault...

**Comment by robimarko** [comment]
Can you share the log?

**Comment by Ansuel** [comment]
@robimarko i was just missing the file from ipq807x target.
Anyway I updated the package to 12.0 but can't update the other as i can't test them...

**Comment by robimarko** [comment]
Thats fine, others dont work as is anyway

File: qca/qca-nss-dp/Makefile [hunk 1]
```diff
@@ -38,22 +38,16 @@ EXTRA_CFLAGS+= \
...
-	$(LN) $(NSS_DP_HAL_DIR)/arch/$(CONFIG_TARGET_BOARD)/nss_$(CONFIG_TARGET_BOARD).h \
+	$(LN) $(NSS_DP_HAL_DIR)/soc_ops/$(CONFIG_TARGET_BOARD)/nss_$(CONFIG_TARGET_BOARD).h \
 		$(PKG_BUILD_DIR)/exports/nss_dp_arch.h
 endef
 
-ifeq ($(CONFIG_TARGET_BOARD), "ipq807x")
-    SOC="ipq807x_64"
-else ifeq ($(CONFIG_TARGET_BOARD), "ipq60xx")
-    SOC="ipq60xx_64"
-endif
```
Comment (robimarko) [inline qca/qca-nss-dp/Makefile:49]: Why was this dropepd?

Comment (Ansuel) [reply qca/qca-nss-dp/Makefile:49]: They changed some stuff in the makefile and the target with _x64 cause missing include.
Also the qsdk makefile started to remove _x64 in their feeds makefile

**Comment by Ansuel** [comment]
> But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0

How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ?

File: qca/qca-nss-dp/Makefile (hunk 1 above)
Comment (robimarko) [reply qca/qca-nss-dp/Makefile:49]: Oh, they are finally moving to 64bit only so a special target is not needed.

**Comment by robimarko** [comment]
> > But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0
> 
> How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ? (and wait for user to complain as nothing works cause they didn't update the nss_packages feed)

Just update the nodes and that's it, at this point I couldn't care less about random users

**Comment by Ansuel** [comment]
Can't find where they changed the clock and reset parsing

**Comment by robimarko** [comment]
Ahh, it looks like they only messed with EDMA v2

**Comment by Boos4721** [comment]
btw, this patche should remove https://github.com/robimarko/nss-packages/blob/qsdk-11.5/qca/qca-ssdk/patches/0003-add-aquantia-phy-id-113CB0-0x31c31C12.patch

https://git.codelinaro.org/clo/qsdk/oss/lklm/qca-ssdk/-/blob/NHSS.QSDK.12.0/src/hsl/phy/hsl_phy.c#L251
https://git.codelinaro.org/clo/qsdk/oss/lklm/qca-ssdk/-/blob/NHSS.QSDK.12.0/include/hsl/phy/hsl_phy.h#L587

**Comment by robimarko** [comment]
Nice, we can then get rid of that patch for AQR

**Comment by 1582130940** [comment]
In testing with 5.10, these were found:
Need to fix 0006-fix-mdio-probe-on-ipq806x.patch to fix mdio for ipq806x;
Need to delete 0008-add-aquantia-phy-id-113CB0.patch, because QSDK12 has full AQR113C B0/B1 support;
The above can be referred to: https://github.com/1582130940/OpenWrt/commit/5a6855405077fe9d178e72ce38bdc169139685ba;
Since both Linux kernel 5.10 and 5.15 are missing socinfo.h (either include/soc/qcom or drivers/soc/qcom), qca-ssdk cannot be compiled.

**Comment by robimarko** [comment]
Generic 5.15 doesnt have socinfo, but IPQ807x tree to which this is targeted does have it.
5.10 is irrelevant at this point really.

**Comment by 1582130940** [comment]
> Generic 5.15 doesnt have socinfo, but IPQ807x tree to which this is targeted does have it. 5.10 is irrelevant at this point really.

```c
#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,0))
#include <soc/qcom/socinfo.h>
#endif
```
But this socinfo.h is for Linux 5.4.0 and above. I don't know what the consequences will be after removing this.

**Comment by robimarko** [comment]
Well, its cause its all written for the QCA 5.4 kernel, no need for removing it

**Comment by robimarko** [comment]
@Ansuel Can you drop that AQR ID patch?
So that we can merge this

**Comment by map-b** [comment]
It is working on AX3600 after removing qca-nss-ssdk 0003 patch and adding and fixing qca-nss-drv patch 0004 from DMA-WIP branch. (Not touched EDMA or switch nodes.)
https://github.com/map-b/nss-packages/commits/qsdk-11.5

**Comment by cc0817** [comment]
guys, any abt ecm on kernel 5.15 news?

**Comment by cc0817** [comment]
> guys, any news abt ecm on kernel 5.15 ?

**Comment by 1582130940** [comment]
In addition, I also found that PKG_SOURCE_DATE (2022-5-16?) and PKG_MIRROR_HASH of nss-firmware are wrong, please refer to my commit: https://github.com/1582130940/OpenWrt/commit/9b24b5229ae5b867e7587cf3ae38586eaa9a3c4e

**Comment by 1582130940** [comment]
After putting socinfo.h in the Linux 5.10 file and fixing the compilation problems caused by the two 5.15 patches, QSDK12.0 has been successfully run under the AX3600.
You can refer to:
https://github.com/1582130940/OpenWrt/commit/d52dbedb45e514f6545d2d506f00b956616cb35b
Also, don't know if it's because the 5.15 patch is not compatible with 5.10, 0009-switchdev-use-new-switchdev-flags.patch and 0005-nss-drv-add-support-for-kernel-5.15.patch patch requires ifdef linux version , can it be fixed?
https://github.com/1582130940/OpenWrt/commit/e5870f3486bd3333a3441c110549c5850b7c2b97

**Comment by rmandrad** [comment]
> > > But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0
> > 
> > 
> > How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ? (and wait for user to complain as nothing works cause they didn't update the nss_packages feed)
> 
> Just update the nodes and that's it, at this point I couldn't care less about random users which dont follow the development and expect a stable product



> > > But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0
> > 
> > 
> > How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ? (and wait for user to complain as nothing works cause they didn't update the nss_packages feed)
> 
> Just update the nodes and that's it, at this point I couldn't care less about random users which dont follow the development and expect a stable product

**Comment by Ansuel** [comment]
@robimarko Sorry for the delay think i will update this today with the removed patch and update the makefile.

One think i can't understand is why sometimes ssdk recompile for no reason...

**Comment by rmandrad** [comment]
b

> > But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0
> 
> How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ? (and wait for user to complain as nothing works cause they didn't update the nss_packages feed)

@Ansuel  great remark about worrying users and worrying about their experience

**Comment by Ansuel** [comment]
> great remark about worrying users and worrying about their experience

the problem is only relevant for new soc... ipq8074 doesn't require any change so we are safe :D user that compile stuff themself can just update the feed with nothing to change...

**Comment by rmandrad** [comment]
> > But, note that EDMA and switch nodes must be updated as they have changed clock and reset parsing in 12.0
> 
> How this should be handled by the user? Just drop a message in the topic and refresh the dtsi in our repo ? (and wait for user to complain as nothing works cause they didn't update the nss_packages feed)

great remark ! any design had to have the user at center ...

**Comment by Ansuel** [comment]
Well it was good to leave an alert but these are complex stuff and not official so it should be user interest to check all the things and try to find a fix themself. I learn the bad way what happen when you give basic user task than only expert user should accomplish (backslash with random flame... private message spamming and so on...)

**Comment by rmandrad** [comment]
> Well it was good to leave an alert but these are complex stuff and not official so it should be user interest to check all the things and try to find a fix themself. I learn the bad way what happen when you give basic user task than only expert user should accomplish (backslash with random flame... private message spamming and so on...)
> 
> (just my 2 cent about the problem about user handling for this kind of thing)
 work in progress (wip)!

**Comment by Ansuel** [comment]
@robimarko should we wait for a final test just to make sure or we can merge?

**Comment by kirdesde** [comment]
@Ansuel : the qca-nss-drv is still missing an updated 0004-nss-drv-rework-NSS_CORE_DMA_CACHE_MAINT-ops.patch from the wip branch and therefore can't be compiled:

`nss_core.h:116:17: error: implicit declaration of function 'dmac_inv_range' [-Werror=implicit-function-declaration]
  116 |                 dmac_inv_range(start, start + size);
      |                 ^~~~~~~~~~~~~~
/home/dirk/openwrt_robi/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-drv-2022-03-07-397c88cf/nss_core.h:119:17: error: implicit declaration of function 'dmac_clean_range' [-Werror=implicit-function-declaration]
  119 |                 dmac_clean_range(start, start + size);
      |                 ^~~~~~~~~~~~~~~~
/home/dirk/openwrt_robi/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-drv-2022-03-07-397c88cf/nss_core.h:122:17: error: implicit declaration of function 'dmac_flush_range' [-Werror=implicit-function-declaration]
  122 |                 dmac_flush_range(start, start + size);
      |                 ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors`

see https://github.com/robimarko/nss-packages/pull/13#issuecomment-1121498210

**Comment by Ansuel** [comment]
@kirdesde we still can't trust the DMA cache change so for now I think it would be wrong to include it here... It will be handled later with another project I'm working on...

**Comment by robimarko** [comment]
@Ansuel Since NSS-DRV doesn't work now, feel free to add the DMA patches to this PR and will get it merged

**Comment by robimarko** [comment]
@Ansuel Just ping me when you have pushed everything you want

**Comment by Ansuel** [comment]
@robimarko this should be all. We can wait for @kirdesde if we want a second check about this

**Comment by robimarko** [comment]
If he wants to try it out before, I am fine with that

**Comment by kirdesde** [comment]
Thanks Ansuel, it's successfully compiling now.

**Comment by Ansuel** [comment]
@robimarko think we can merge this :D

**Comment by zxlhhyccc** [comment]
The following error occurs in the system log:
image

**Comment by kirdesde** [comment]
> The following error occurs in the system log: image

No, this is not an error but just an info message. kern **info** kernel

**Comment by zxlhhyccc** [comment]
ecm-wip:qca-nss-ecm compiler error：
```
neric/qca-nss-ecm-2022-05-01-a063d80c/ecm_classifier_default.o
  CC [M]  /home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o
/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.c: In function 'ecm_interface_list_stats_update':
/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.c:6546:33: error: implicit declaration of function 'br_fdb_entry_refresh' [-Werror=implicit-function-declaration]
 6546 |                                 br_fdb_entry_refresh(dev, mac_addr, 0);
      |                                 ^~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
scripts/Makefile.build:277: recipe for target '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o' failed
make[4]: *** [/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o] Error 1
Makefile:1868: recipe for target '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c' failed
make[3]: *** [/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c] Error 2
make[3]: Leaving directory '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.38'
````

**Comment by kirdesde** [comment]
> ecm-wip:qca-nss-ecm compiler error：
> 
> ```
> neric/qca-nss-ecm-2022-05-01-a063d80c/ecm_classifier_default.o
>   CC [M]  /home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o
> /home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.c: In function 'ecm_interface_list_stats_update':
> /home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.c:6546:33: error: implicit declaration of function 'br_fdb_entry_refresh' [-Werror=implicit-function-declaration]
>  6546 |                                 br_fdb_entry_refresh(dev, mac_addr, 0);
>       |                                 ^~~~~~~~~~~~~~~~~~~~
> cc1: all warnings being treated as errors
> scripts/Makefile.build:277: recipe for target '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o' failed
> make[4]: *** [/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c/ecm_interface.o] Error 1
> Makefile:1868: recipe for target '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c' failed
> make[3]: *** [/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/qca-nss-ecm-2022-05-01-a063d80c] Error 2
> make[3]: Leaving directory '/home/lin/ax6/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.38'
> ```

ECM is not in the focus of the PR.

**Pull request merged**